packages/core/src/indexing/local-index-api.ts

```diff
@@ -20,6 +21,14 @@ export class LocalIndexApi implements IndexApi {
     private readonly logger: DiagnosticsLogger
   ) {}
 
+  /**
+   * Add stream to index in appropriate model table
+   * @param args
+   */
+  async indexStream(args: IndexStreamArgs): Promise<void> {
```

Comment:
IMO this method should handle the check https://github.com/ceramicnetwork/js-ceramic/pull/2252/files#diff-0625065ede662c738d45fe3f49a8a1afaba805c6208872cdbf82859722562e5cR47

Let's leave the SqliteIndexAPI to deal with the databases only.

IMO, when we add an intermediate table for StreamID -> table, this `LocalIndexAPI` should be a place to maintain an inmemory cache of it.

Reply:
Hmm... I agree about moving the check of whether or not the stream should be indexed up here, that way it only needs to live in one place even when we have multiple index database implementations.

As for the cache of the mapping from Model streamid to table name, that part I think should live at the lower level in the database-specific index implementation.  This is because we might use a different system to maintain the streamid->table mapping for different database implementations.

Reply:
Sounds good. I'll move the check as suggested by Sergey for the time being. For the cache that makes sense going forward.